src/gui/tray/usermodel.cpp

```diff
@@ -116,11 +116,37 @@ void User::slotBuildNotificationDisplay(const ActivityList &list)
 {
     _activityModel->clearNotifications();
 
-    foreach (auto activity, list) {
+    const auto multipleAccounts = AccountManager::instance()->accounts().count() > 1;
+    ActivityList toNotifyList;
+
+    for(const auto &activity : list) {
         if (_blacklistedNotifications.contains(activity)) {
             qCInfo(lcActivity) << "Activity in blacklist, skip";
             continue;
         }
+
+        if(_notifiedNotifications.contains(activity._id)) {
+            qCInfo(lcActivity) << "Activity already notified, skip";
+            continue;
+        }
+
+        toNotifyList.append(activity);
+    }
```

Comment:
maybe you could simply use `std::copy_if` to make it easy to read and maybe less verbose